msgraph/models.go

```diff
@@ -730,6 +731,28 @@ func (c ConditionalAccessGuestsOrExternalUsers) MarshalJSON() ([]byte, error) {
 		GuestOrExternalUserTypes:               val,
 		conditionalAccessGuestsOrExternalUsers: (*conditionalAccessGuestsOrExternalUsers)(&c),
 	}
+
+	const externalTenantsTypeAll = "#microsoft.graph.conditionalAccessAllExternalTenants"
+	const externalTenantsTypeEnumerated = "#microsoft.graph.conditionalAccessEnumeratedExternalTenants"
+	setExternalTenantsObjectType := func(c *conditionalAccessGuestsOrExternalUsers) {
+		if c == nil {
+			return
+		}
+		if c.ExternalTenants == nil {
+			return
+		}
+		if c.ExternalTenants.MembershipKind == nil {
+			return
+		}
+		switch *c.ExternalTenants.MembershipKind {
+		case ConditionalAccessExternalTenantsMembershipKindAll:
+			c.ExternalTenants.ODataType = utils.StringPtr(externalTenantsTypeAll)
+		case ConditionalAccessExternalTenantsMembershipKindEnumerated:
+			c.ExternalTenants.ODataType = utils.StringPtr(externalTenantsTypeEnumerated)
+		}
+	}
+	setExternalTenantsObjectType(guestOrExternalUsers.conditionalAccessGuestsOrExternalUsers)
```

Comment:
As there's no need for reusability here, could we simplify this by forgoing the constant declarations and move the `if`+`switch` blocks out into the parent method, dropping the anonymous func? The anonymous struct embeds a pointer to the original struct so this should work?

```suggestion
	if c.ExternalTenants != nil && c.ExternalTenants.MembershipKind != nil {
		switch *c.ExternalTenants.MembershipKind {
		case ConditionalAccessExternalTenantsMembershipKindAll:
			c.ExternalTenants.ODataType = utils.StringPtr("#microsoft.graph.conditionalAccessAllExternalTenants")
		case ConditionalAccessExternalTenantsMembershipKindEnumerated:
			c.ExternalTenants.ODataType = utils.StringPtr("#microsoft.graph.conditionalAccessEnumeratedExternalTenants")
		}
	}
```